### PR TITLE
MGDOBR-836: mask sensitive parameters in RHOSE user APIs

### PR DESCRIPTION
**Pull request by kostola**
[MGDOBR-836](https://issues.redhat.com/browse/MGDOBR-836)

# Description

First PR related to the [MGDOBR-59](https://issues.redhat.com/browse/MGDOBR-59) epic.

**IMPORTANT:** his focuses only on *how the user API responds* in case of sensitive parameters. There's *no logic to store them* in a dedicated vault, they're still stored in the database.

## How It Works

- The sensitive parameters corresponds to the JSON schema properties containing `"format": "password"` in the definition.
- The detection logic is implemented in `ProcessorCatalogService#getActionPasswordProperties` and `ProcessorCatalogService#getSourcePasswordProperties`
- `GatewaySecretsHandler` contains the generic mask logic
- `ProcessorServiceImpl` masks the fields in `ProcessorResponse` in the `toResponse` method.
- `ProcessorServiceImpl` needs to unmask the fields when trying to update the processor. I added comments to the tricky parts in the code.

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Comment by Srihari1192** [comment]
@kostola 
I have one query with respect to masking the action parameters , for now we are focusing on  slack_token , slack_webhook_url parameters but not for Webhook endpoint. I think we should mask this Parameter as well.

In case we are handling mask for Webhook endpoint parameters means , we need to be taken care for Webhook error handler as well

File: manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java
Comment (robpblake) [inline manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java:246]: @kostola Just wondering if this method is better placed on the `GatewaySecretsHandler` class instead, to centralise the mask/unmask logic.

**Review by manstis — CHANGES_REQUESTED** [review]
Can you please also update [this](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java#L216-L226) now it is merged?

You will need to rebase this PR.

**Comment by r00ta** [comment]
Please do not merge this until I remove the label -> I'm checking that all the knative changes are working on main/dev

File: manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java
Comment (manstis) [inline manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java:201]: Do you also need to update [these](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/fa09fc86ad2c9f6146b2ace88970d61415555709/manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java#L240-L241) lines to use the unmasked values?

Comment (kostola) [reply manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java]: I just fixed the update :+1:

**Comment by kostola** [comment]
> Can you please also update [this](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java#L216-L226) now it is merged?
> 
> You will need to rebase this PR.

I rebased the PR and updated everything :+1:

File: manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java
Comment (kostola) [reply manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java]: I renamed this method because the name was misleading. This is not actually an unmasking, because the definition in this PR is always unmasked in the database. For this reason I don't think it's useful to add an `unmask` method to `GatewaySecretsHandler` at the moment (it is required in the next PR where I will interact with the AWS Secrets Manager).
What I am doing here is merging the existing secrets values for fields that are unchanged inside the request, so that the following checks perform correctly (e.g. if the processor is unchanged nothing is actually done). Without the merge, the checks would always fail because the unchanged secrets values are missing in the request.

File: processor/configuration/src/main/java/com/redhat/service/smartevents/processor/GatewaySecretsHandler.java
Comment (manstis) [inline processor/configuration/src/main/java/com/redhat/service/smartevents/processor/GatewaySecretsHandler.java:26]: "..password properties.." isn't very generic for a general "secrets" service.

Your tests show User Name and Token as two other types of secret. Did you want to consider changing the method name?

File: processor/configuration/src/main/java/com/redhat/service/smartevents/processor/GatewaySecretsHandler.java
Comment (manstis) [inline processor/configuration/src/main/java/com/redhat/service/smartevents/processor/GatewaySecretsHandler.java:31]: Likewise.

**Review by manstis — APPROVED** [review]

**Review by Srihari1192 — APPROVED** [review]

**Comment by kostola** [comment]
@kelvah since this change impacts the frontend, I'll wait for your final approval before merging :+1:

**Comment by sthundat** [comment]
Hi @kostola ,
I have created a `kafka_topic_sink_0.1` processor  using `KafkaTopic` action.
And the response I am seeing the sensitive parameters like below,

            "kafka_client_id": "srvc-acct-df06dd9b-e7d4-4e25-b257-f7a96f24ec3e",
            "kafka_client_secret": "1e738ef8-fbd7-4a29-adb4-1362264b36ea".

These values should be masked.
WDYT?
Thanks,
Savitha

**Review by sthundat — APPROVED** [review]
It was a quick fix @kostola  ,it looks good to me

**Comment by vpellegrino** [comment]
@kelvah worked on that part on the UI side, so I would prefer to wait for his return (following Monday) before approving this PR.
@kostola, please let me know if you need a quick review of it or if it is OK to wait for him.

**Review by kelvah — APPROVED** [review]

**Pull request merged**
